AdobeStockClient/Model/SearchParametersProvider/Premium.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Magento\AdobeStockClient\Model\SearchParametersProvider;
+
+use AdobeStock\Api\Models\SearchParameters;
+use Magento\AdobeStockClientApi\Api\SearchParameterProviderInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+
+class Premium implements SearchParameterProviderInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function apply(SearchCriteriaInterface $searchCriteria, SearchParameters $searchParams): SearchParameters
+    {
+        foreach ($searchCriteria->getFilterGroups() as $filterGroup) {
```

Comment:
I would propose to extract all blocks of
```php
foreach ($searchCriteria->getFilterGroups() as $filterGroup) {
            foreach ($filterGroup->getFilters() as $filter) {
                if ($filter->getField() === 'premium_price_filter') {
                    $searchParams->setFilterPremium($filter->getValue());
                    return $searchParams;
                }
            }
        }
```

To a separate class as there are similar and the difference only in field name and search parameter setter method's name.

Reply:
Done